src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,242 @@
+        from_spherical(-std::cos(theta), sample_phi(rng)), inc_direction_);
+
+    // Outgoing secondaries are electron and positron
+    electron_pair[0].def_id = shared_.electron_id;
+    electron_pair[1].def_id = shared_.positron_id;
+    // Energies of secondaries shared equally
+    electron_pair[0].energy
+        = units::MevEnergy{0.5 * epsilon * inc_energy_.value()};
+    electron_pair[1].def_id
+        = units::MevEnergy{0.5 * epsilon * inc_energy_.value()};
+    // Calculate produced pair directions via conservation of momentum
+    for (unsigned int p = 0; p < allocate_.capacity(); p++)
+    {
+        for (int i = 0; i < 3; ++i)
+        {
+            electron_pair[p].direction[i]
+                = inc_direction_[i] * inc_energy_.value()
+                  - electron_pair[p].direction[i]
+                        * electron_pair[p].energy.value();
+        }
+        normalize_direction(&electron_pair[p].direction);
+    }
+
+    return result;
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::delta(size_type Z,
+                                                       real_type eps) const
+{
+    return 136 * std::pow(Z, -1 / 3) * epsilon0_ * epsilon0_
+           / (eps * (1 - eps));
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::Phi1(real_type delta) const
+{
+    if (delta <= 1)
+    {
+        return 20.867 - 3.242 * delta + 0.625 * delta * delta;
+    }
+    return Phi12(delta);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::Phi2(real_type delta) const
+{
+    if (delta <= 1)
+    {
+        return 20.209 - 1.930 * delta - 0.086 * delta * delta;
+    }
+    return Phi12(delta);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::Phi12(real_type delta) const
+{
+    return 21.12 - 4.184 * std::log(delta + 0.952);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::CoulombCorr(size_type Z) const
+{
+    real_type term1 = 8 / 3 * std::log(Z);
+
+    if (inc_energy_.value() < 50.0) // 50 MeV
+    {
+        return term1;
+    }
+
+    return term1 + 8 * this->CoulombCorr_aux(Z);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::CoulombCorr_aux(size_type Z) const
+{
+    real_type alphaZ_sq = constants::alpha_fine_structure
+                          * constants::alpha_fine_structure * Z * Z;
+    real_type alphaZ_sqsq = alphaZ_sq * alphaZ_sq;
+    return alphaZ_sq
+           * (1 / (1 + alphaZ_sq) + 0.20206 - 0.0369 * alphaZ_sq
+              + 0.0083 * alphaZ_sqsq - 0.0020 * alphaZ_sq * alphaZ_sqsq);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::Phi1_aux(real_type delta,
+                                                          size_type Z) const
+{
+    return (3 * this->Phi1(delta) - this->Phi2(delta) - this->CoulombCorr(Z));
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::Phi2_aux(real_type delta,
+                                                          size_type Z) const
+{
+    return (3 / 2 * this->Phi1(delta) - 0.5 * this->Phi2(delta)
```

Comment:
`3 / 2 * x` is `1 * x` is `x` -- write as `1.5` or declare a local `constexpr real_type three_halves = 3.0 / 2;`.